apis/metal3.io/v1alpha1/baremetalhost_types.go

```diff
@@ -278,6 +278,14 @@ type BareMetalHostSpec struct {
 	// the power status and hardware inventory inspection. If the
 	// Image field is filled in, this field is ignored.
 	ExternallyProvisioned bool `json:"externallyProvisioned,omitempty"`
+
+	// When set to None, automated cleaning will be avoided
+	// during provisioning and deprovisioning.
+	// +optional
+	// +kubebuilder:default:=Full
+	// +kubebuilder:validation:Enum:=Full;Disabled
+	// +kubebuilder:validation:Optional
+	AutomatedCleaningMode string `json:"automatedCleaningMode,omitempty"`
```

Comment:
For `BootMode` we define a type and set the kubebuilder options on that type. Can we do the same for this enum? That would let us have `const` values for comparisons, too.

Reply:
Yes of course. Fixed.